File: backend/DormitoryManagement.Data/Models/Registration.cs
using System.ComponentModel.DataAnnotations;

namespace DormitoryManagement.Data.Models
{
    public partial class Registration
    {
        public int RegistrationId { get; set; }
        [MaxLength(50)]
        public string StudentNumber { get; set; }
        [MaxLength(50)]
        public string RoomNumber { get; set; }
        public DateOnly? CheckInDate { get; set; }
        public DateOnly? CheckOutDate { get; set; }

        public Student Student { get; set; }
        public Room Room { get; set; }
    }
}

File: frontend/src/app/models/Registration.ts
export interface Registration {
  expanded: boolean;
  registrationId: number;
  studentNumber: string;
  roomNumber: string;
  checkInDate?: string;
  checkOutDate?: string;
}

export interface RegistrationDTO {
  registrationId?: number;
  studentNumber: string;
  roomNumber: string;
  checkInDate?: string;
  checkOutDate?: string;
  expanded: boolean;
}
